test/asyncpromise.cc

```diff
@@ -0,0 +1,67 @@
+#include "napi.h"
+
+using namespace Napi;
+
+class TestPromise : public AsyncPromise<TestPromise>::BaseTask {
+public:
+  enum class Test
+  {
+    succeed_before_queue = 1,
+    succeed_in_execute,
+    fail_before_queue,
+    fail_in_execute
+  };
+
+  static Value MakePromise(const CallbackInfo& info) {
+    auto test = (Test)info[0].As<Number>().Uint32Value();
+    Object resource = info[1].As<Object>();
+    std::string data = info[2].As<String>();
+
+    AsyncPromise<TestPromise> async_promise(info.Env(), "TestResource", resource);
+
+    async_promise.Task()->_data = data;
+
+    switch (test)
+    {
+    case Test::succeed_before_queue:
```

Comment:
This should be indented I think to line up with the other cases.

Reply:
Thanks, Michael. Now fixed.